### ISSUE TITLE
[BUG] The org.opensearch.remotestore.CreateRemoteIndexIT.testDefaultRemoteStoreNoUserOverride is flaky

### ISSUE THREAD

**Issue by reta** [opensearch-project/OpenSearch#12229]
### Describe the bug

The test case `org.opensearch.remotestore.CreateRemoteIndexIT.testDefaultRemoteStoreNoUserOverride` is flaky:

```
org.opensearch.remotestore.CreateRemoteIndexIT.testDefaultRemoteStoreNoUserOverride

java.lang.AssertionError: shard [test-idx-1][0] is not locked
	at __randomizedtesting.SeedInfo.seed([AC3DAFC531F23A2A]:0)
	at org.opensearch.env.NodeEnvironment.deleteShardDirectoryUnderLock(NodeEnvironment.java:579)
	at org.opensearch.indices.IndicesService.deleteShardStore(IndicesService.java:1233)
	at org.opensearch.index.IndexService.onShardClose(IndexService.java:678)
	at org.opensearch.index.IndexService$StoreCloseListener.accept(IndexService.java:801)
	at org.opensearch.index.IndexService$StoreCloseListener.accept(IndexService.java:788)
	at org.opensearch.index.store.Store.closeInternal(Store.java:573)
	at org.opensearch.index.store.Store$1.closeInternal(Store.java:194)
	at org.opensearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:78)
	at org.opensearch.index.store.Store.decRef(Store.java:546)
	at org.opensearch.index.engine.InternalEngine.refresh(InternalEngine.java:1868)
	at org.opensearch.index.engine.InternalEngine.maybeRefresh(InternalEngine.java:1844)
	at org.opensearch.index.shard.IndexShard.scheduledRefresh(IndexShard.java:4511)
	at org.opensearch.index.IndexService.maybeRefreshEngine(IndexService.java:1026)
	at org.opensearch.index.IndexService$AsyncRefreshTask.runInternal(IndexService.java:1170)
	at org.opensearch.common.util.concurrent.AbstractAsyncTask.run(AbstractAsyncTask.java:159)
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:854)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Related component

Search

### To Reproduce

```
 ./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.CreateRemoteIndexIT.testDefaultRemoteStoreNoUserOverride" -Dtests.seed=AC3DAFC531F23A2A
```

### Expected behavior

The test must always pass

### Additional Details

**Plugins**
Standard

**Screenshots**

**Host/Environment (please complete the following information):**
 - CI

**Additional context**
- https://build.ci.opensearch.org/job/gradle-check/33509/testReport/junit/org.opensearch.remotestore/CreateRemoteIndexIT/testDefaultRemoteStoreNoUserOverride/

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/dbwiddis) [3](https://github.com/andrross)]
@reta Thanks for filing, looking forward to a PR to resolve this.

**Comment by ltaragi** [comment]
Fixed by #14601
